src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java

```diff
@@ -122,8 +123,8 @@ protected void writeTo(OutputStream os) throws IOException {
         }
     }
 
-    protected void writeTo(OutputStream os, ContentFilter filter) throws IOException {
-        if (isBinary || filter == null) {
```

Comment:
Dropping this condition was actually the reason for https://github.com/jenkinsci/support-core-plugin/pull/502/files#r1491768009, see https://github.com/jenkinsci/support-core-plugin/pull/509.

Reply:
Well, only partially. Previously we would have gone through the `writeTo(OutputStream)` overload above instead. If this was the only issue, then we would have noticed the lack of `return` causing most types of file content to be duplicated!